### DISCUSSION TITLE
Edit Search phrase

### DISCUSSION THREAD

**Discussion by Mepcuk** [orchidjs/tom-select#231] [Q&A] [answered]
I have a problem using TomSelect. I use select data update via API, get data but it is not displayed, because it is try to search keyword which not exist. I need to search word (exact equal word) and word* (like). At backend i was created API endpoint which send correct values, however it not displayed *reason.

`  new TomSelect('#tom-select-it',{
valueField: 'id',
labelField: 'title',
// searchField: 'title',
plugins: ['remove_button'],
// fetch remote data
load: function(query, callback) {

   var url = '/display/group/?search=' + encodeURIComponent(query);
   fetch(url)
      .then(response => response.json())
      .then(json => {
          console.log(json);
         callback(json.items);
      }).catch(()=>{
         callback();
      });

},`

 I console panel i receive response.
`{
"total_count": 1,
"incomplete_results": false,
"items": [
{
"title": "0122 Zāles- Tējas- -",
"value": 1301,
"score": 1
},
{
"title": "0122 Zāles- Tējas- -",
"value": 1301,
"score": 1
}
]
}`

Did not understand what i missing......

**Comment by oyejorge** [comment]
Your configuration has ```valueField: 'id',``` but the data returned from your server doesn't have any ```id``` fields.

**Reply by Mepcuk** [reply]
> Your configuration has `valueField: 'id',` but the data returned from your server doesn't have any `id` fields.

Problem described below

**Comment by Mepcuk** [comment]
@oyejorge  Thanks for findin mistake however it not solve problem

when i type 0101* (response from backend 
0: {title: '01010101 Zāles- Pretsāpju līdzekļi- Tabletes- Ibuprofēns', value: 1204, score: 1}
1: {title: '01010101 Zāles- Pretsāpju līdzekļi- Tabletes- Ibuprofēns', value: 1204, score: 1}
2: {title: '01010102 Zāles- Pretsāpju līdzekļi- Tabletes- Paracetamols', value: 1205, score: 1})
at backend i filtered data by
`            return $this->createQueryBuilder('d')
                ->andWhere("d.display_group_id LIKE :slug OR d.display_group LIKE :slug")
                ->setParameter('slug', '%'.$slug.'%')
                ->getQuery()
                ->getResult();`

data successtully received no problem at backend but not displayed because as i correct understand filtered second time byTomSelect....

**Comment by Mepcuk** [comment, marked as answer]
Solved: example typing in search 01*

in front-end added another search field - search field
new TomSelect('#tom-select-it',{ valueField: 'value', labelField: 'title', searchField: 'search',

at back-end - when receive 01* (LIKE request) added to matched results field search with value "01*" but if i search directly 01 in matched results search added title
